### PR TITLE
chore: Checking for more instances of EagerTensors [DET-4566]

### PR DESCRIPTION
**Pull request by mackrorysd**
## Description

This was first done in DET-4299 / DET-4202 but with more recent versions of TensorFlow, the same thing is being done in some metrics too.

This makes such cases not fatal (as they would otherwise fail to serialize), but I'm wondering if we should actually also use .asscalar(). If Horovod is promoting scalars to tensors, a NumPy array doesn't entirely reverse it. On the other hand, if we were to somehow get a numpy array as a metric (if we supported arbitrary, non-scaler metric types) this would also not be ideal.

### CODE REVIEW

File: harness/determined/keras/_tf_keras_trial.py [hunk 1]
```diff
@@ -802,10 +802,20 @@ def _allreduce_logs(self, logs: Dict) -> Dict:
         logging.debug(f"all-reducing logs on worker {hvd.rank()} for {len(keys)} keys {keys}.")
         return {key: np.array(hvd.allreduce(logs[key], name=key)) for key in keys}
 
+    def _convertPossibleTensor(self, possibleTensor: Any) -> Any:
+        if isinstance(possibleTensor, EagerTensor):
+            # Horovod will promote an int to a tensor in eager mode.
```
Comment (rb-determined-ai) [inline harness/determined/keras/_tf_keras_trial.py:807]: probably worth re-writing this comment to be up-to-date, as it seems horovod is no longer the only source of this problem, right?

File: harness/determined/keras/_tf_keras_trial.py [hunk 2]
```diff
@@ -802,10 +802,20 @@ def _allreduce_logs(self, logs: Dict) -> Dict:
         logging.debug(f"all-reducing logs on worker {hvd.rank()} for {len(keys)} keys {keys}.")
         return {key: np.array(hvd.allreduce(logs[key], name=key)) for key in keys}
 
+    def _convertPossibleTensor(self, possibleTensor: Any) -> Any:
```
Comment (rb-determined-ai) [inline harness/determined/keras/_tf_keras_trial.py:805]: please use snake_case for function names in python code

File: harness/determined/keras/_tf_keras_trial.py [hunk 3]
```diff
@@ -802,10 +802,20 @@ def _allreduce_logs(self, logs: Dict) -> Dict:
...
+        if isinstance(possibleTensor, EagerTensor):
```
Comment (rb-determined-ai) [inline harness/determined/keras/_tf_keras_trial.py:806]: also use snake case for variables (actually anything except classes)

**Review by rb-determined-ai — APPROVED** [review]

File: harness/determined/keras/_tf_keras_trial.py (hunk 1 above)
Comment (mackrorysd) [reply harness/determined/keras/_tf_keras_trial.py:807]: Yeah I wasn't sure if it was Horovod, but then some other change in Tensor triggered it for other things. I wasn't able to find anything definitive myself so I trusted your comment. If you're also not sure, I'll just make the comment more vague :)

Comment (rb-determined-ai) [reply harness/determined/keras/_tf_keras_trial.py:807]: If we are not sure, it might be nice to leave the original comment in its original location, and put a new comment in the other places we call `_convert_possible_tensor`
